addons/pos_restaurant/static/src/overrides/models/models.js

```diff
@@ -29,7 +29,7 @@ patch(Order.prototype, {
         super.init_from_JSON(...arguments);
         if (this.pos.config.module_pos_restaurant) {
             this.tableId = json.table_id;
-            this.validation_date = moment.utc(json.creation_date).local().toDate();
+            this.validation_date = json.creation_date;
```

Comment:
Can you try "serializeDate" from "@web/core/l10n/dates"? And in the getDate of ticket_screen, you can keep the logic.